swap/src/protocol/bob/event_loop.rs

```diff
@@ -250,11 +251,17 @@ impl EventLoopHandle {
     }
 
     pub async fn request_spot_price(&mut self, btc: bitcoin::Amount) -> Result<monero::Amount> {
-        Ok(self
+        let response = self
             .spot_price
             .send_receive(spot_price::Request { btc })
-            .await?
-            .xmr)
+            .await?;
+
+        match response {
+            Response::Xmr(xmr) => Ok(xmr),
+            Response::Error(error) => {
+                bail!(error);
+            }
+        }
```

Comment:
That is equivalent to `?` if you would use `Result` instead of `Response`. Not sure if such a good idea though :)